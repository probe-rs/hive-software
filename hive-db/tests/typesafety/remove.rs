use hive_db::{CborDb, HiveDb, Key};

const FLUSH_INTERVAL_MS: u64 = 60_000;
const CACHE_CAPACITY: u64 = 52_428_800;

fn main() {
    let db = HiveDb::open("dummy_path", FLUSH_INTERVAL_MS, CACHE_CAPACITY);

    let tree = db.open_tree("tree");

    let my_key: Key<u8> = Key::new("NiceNumber");

    let wrong_type: Option<()> = tree.c_remove(&my_key).unwrap();
}
