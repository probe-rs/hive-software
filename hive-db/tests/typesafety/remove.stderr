error[E0308]: mismatched types
  --> tests/typesafety/remove.rs:13:34
   |
13 |     let wrong_type: Option<()> = tree.b_remove(&my_key).unwrap();
   |                     ----------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `u8`
   |                     |
   |                     expected due to this
   |
   = note: expected enum `Option<()>`
              found enum `Option<u8>`
